src/api/store/userprofile.py

```diff
@@ -575,62 +575,42 @@ def handle_csv(self, context: Context, args, temporarylocation) -> (dict, MassEn
-              message = cadmin.full_name + " invited you to join the following MassEnergize Community: " + registered_community.name + "\n"
-              mess = args.get('message', None)
-              if mess and mess != "":
-                message += "They have included a message for you here:\n"
-                message += mess
-              if team:
-                message += "You have been assigned to the following team: " + team.name + "\n"
-              link = "massenergize.org/" + str(registered_community.subdomain) + "/signup"
-              message += "Use the following link to join " + registered_community.name + ": " + link
-              send_massenergize_email(subject= cadmin.full_name + " invited you to join a MassEnergize Community", msg=message, to=new_user.email)
-            else:   
-              if reader.line_num != 0:
-                invalid_emails.append(reader.line_num) 
-          except Exception as e:
-            print(str(e))
-            return None, CustomMassenergizeError(e)
-      res = {'invalidEmails' : invalid_emails}
-      return res
+      user = UserProfile.objects.filter(email=csv_row[email_field]).first()
+      if not user:
+        if csv_row[email_field] == "" or not csv_row[email_field]:
+          return None, CustomMassenergizeError("One of more of your user(s) lacks a valid email address. Please make sure all your users have valid email addresses listed.")
+        new_user: UserProfile = UserProfile.objects.create(
+          full_name = csv_row[first_name_field] + ' ' + csv_row[last_name_field], 
+          preferred_name = csv_row[first_name_field], 
+          email = csv_row[email_field],
+          is_vendor = False, 
+          accepts_terms_and_conditions = False
+        )
+        new_user.save() 
+        if registered_community:
+          new_user.communities.add(registered_community)
+      else: 
+        new_user: UserProfile = user  
+      team_name = args.get('team_name', None)
+      team = None
+      if team_name != "none":
+        team = Team.objects.filter(name=team_name).first()
+        team.members.add(new_user)
+        team.save()
+      new_user.save()
+      # send email inviting user to complete their profile
+      message = cadmin.full_name + " invited you to join the following MassEnergize Community: " + registered_community.name + "\n"
+      mess = args.get('message', None)
+      if mess and mess != "":
+        message += "They have included a message for you here:\n"
+        message += mess
+      if team:
+        message += "You have been assigned to the following team: " + team.name + "\n"
+      link = "massenergize.org/" + str(registered_community.subdomain) + "/signup"
+      message += "Use the following link to join " + registered_community.name + ": " + link
+      send_massenergize_email(subject= cadmin.full_name + " invited you to join a MassEnergize Community", msg=message, to=new_user.email)
```

Comment:
we should really not be doing any  business logic things inside the store file.  We've done very poorly in the past and i think we have a lot of tech debt in moving all those logic from the store files into the service files.  

In this scenario, call the store to just save data or fetch data and that's it.  Dont send emails in store, don't do multiple things in the same function too.  Functions should usually do one thing and only one thing.